web/packages/teleport/src/UnifiedResources/Resources.tsx

```diff
@@ -0,0 +1,86 @@
+/*
+Copyright 2019-2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React from 'react';
+import { Flex, Indicator, Box, ButtonPrimary } from 'design';
+
+import styled from 'styled-components';
+
+import {
+  FeatureBox,
+  FeatureHeader,
+  FeatureHeaderTitle,
+} from 'teleport/components/Layout';
+import ErrorMessage from 'teleport/components/AgentErrorMessage';
+import useTeleport from 'teleport/useTeleport';
+
+import { useResources } from './useResources';
+import { ResourceCard } from './ResourceCard';
+
+export function Resources() {
+  const teleCtx = useTeleport();
+  const { attempt, fetchedData, fetchMore } = useResources(teleCtx);
+  const observed = React.useRef(null);
+
+  React.useEffect(() => {
+    if (observed.current) {
+      const observer = new IntersectionObserver(entries => {
+        console.log('INTERSECT', entries[0]);
+        if (entries[0].isIntersecting) {
```

Comment:
I've never used an `IntersectionObserver` before - is `entries` guaranteed to be non-empty?

Reply:
According to MDN, "The list of entries received by the callback includes one entry for each target which reported a change in its intersection status". However, it won't hurt to protect from getting an empty array here, so I'm doing it.